neo/rawio/blackrockrawio.py

```diff
@@ -971,32 +982,44 @@ def __read_nsx_dataheader_variant_b(
         Reads the nsx data header for each data block following the offset of
         file spec 2.2, 2.3, and 3.0.
         """
-        filename = ".".join([self._filenames["nsx"], f"ns{nsx_nb}"])
+        filename = f"{self._filenames['nsx']}.ns{nsx_nb}"
 
         filesize = self.__get_file_size(filename)
 
         data_header = {}
-        index = 0
 
-        if offset is None:
-            offset = self.__nsx_basic_header[nsx_nb]["bytes_in_headers"]
+        offset_to_first_data_block = offset or int(self.__nsx_basic_header[nsx_nb]["bytes_in_headers"])
```

Comment:
I think this will still evaluate to the second condition even if someone explicitly wanted an offset of 0.
If you really want a one-liner, it would be `offset if offset is not None else ...`

Reply:
Also the input offset needs to be converted to `int`

Reply:
Ah, thanks, indeed. That clarifies it, I did not read well above.

Reply:
Fixed.